arangod/Agency/Agent.h

```diff
@@ -388,6 +392,11 @@ class Agent : public arangodb::Thread,
   /// For _ioLock: We put in assertions to ensure that when this lock is
   /// acquired we do not have the _tiLock.
 
+  /// Rules for reading and locks
+  ///    _oLock(here)
+  ///    _logLock(in State) _waiForCV (here)
+  /// _oLock guards the _readDB
```

Comment:
This needs more information saying _commitIndex also guarded, OR point to comments about _oLock to see what is guarded.

_waitForCV needs to clarify if its lock is need for _commitIndex reads, or if both oLock and waitForCV needed for writes to _commitIndex